cmd/server/cadence.go

```diff
@@ -73,12 +75,24 @@ func startHandler(c *cli.Context) {
 	}
 
 	services := getServices(c)
+	sigc := make(chan os.Signal, 1)
+	signal.Notify(sigc, syscall.SIGTERM)
 	for _, svc := range services {
 		if _, ok := cfg.Services[svc]; !ok {
 			log.Fatalf("`%v` service missing config", svc)
 		}
 		server := newServer(svc, &cfg)
 		server.Start()
+		go func() {
+			sig := <-sigc
+			switch sig {
+			case syscall.SIGTERM:
+				log.Println("Received SIGTERM signal, initiating shutdown.")
+				server.Stop()
+				os.Exit(0)
```

Comment:
the first goroutine to finish will call Stop() and so, any other service which is in the midst of a Stop() would just terminate forcibly. For this to work correctly, you need to have exactly one sigterm callback (not one per service) and that callback should initiate shutdown of all services when it receives a signal.